### ISSUE TITLE
Error : Unable to Upgrade an existing AKS cluster with Azure Key Vault Provider for Secrets Store CSI Driver support

### ISSUE THREAD

**Issue by ksingh-scogo** [Azure/azure-cli#27240]
### Describe the bug

Following this [documentation](https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver#upgrade-an-existing-aks-cluster-with-azure-key-vault-provider-for-secrets-store-csi-driver-support) to Upgrade an existing AKS cluster with Azure Key Vault Provider for Secrets Store CSI Driver support

Getting this error while running the command `az aks enable-addons --addons azure-keyvault-secrets-provider --name Soooo-Staging-AKS-1 --resource-group Soooo_Staging_RG`

```
cli.azure.cli.core.azclierror: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
az_command_data_logger: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
```

### Related command

```
az aks enable-addons --addons azure-keyvault-secrets-provider --name Soooo-Staging-AKS-1 --resource-group Soooo_Staging_RG
```

### Errors

```
cli.azure.cli.core.azclierror: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
az_command_data_logger: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
```

### Issue script & Debug output

```
cli.azure.cli.core.azclierror: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
az_command_data_logger: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})
```

### Expected behavior

Should be able to successfully Upgrade an existing AKS cluster with Azure Key Vault Provider for Secrets Store CSI Driver support

### Environment Summary

{
  "azure-cli": "2.50.0",
  "azure-cli-core": "2.50.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "aks-preview": "0.5.152"
  }
}

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by navba-MSFT** [comment]
@ksingh-scogo Thanks for reaching out to us and reporting this issue. We are looking into this issue and we will provide an update.

**Comment by FumingZhang** [comment]
Please provide debug output in the issue next time.

I found your [post](https://github.com/MicrosoftDocs/azure-docs/issues/112875#issuecomment-1689386961) in another issue, from error log, the exception happened around

> File "/Users/ksingh/.azure/cliextensions/aks-preview/**azext_aks_preview**/**custom.py**", line **1617**, in aks_enable_addons
>     **ensure_container_insights_for_monitoring**(
>   File "/opt/homebrew/Cellar/azure-cli/2.50.0_1/libexec/lib/python3.10/site-packages/azure/cli/command_modules/**acs**/**addonconfiguration**.py", line **546**, in **ensure_container_insights_for_monitoring**
>     send_raw_request(

To reach [aks-preview/custom.py#1617](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/custom.py#L1617), the existing cluster should have monitoring addon enabled and it's configured to use aad auth.

> azure.cli.core.azclierror.HTTPError: Bad Request({"error":{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable.","details":[{"code":"InvalidPayload","message":"The new DCR kind `` is different from the previous one `Linux`. The kind is immutable."}]}})

The DCR kind in the sent request body is hardcoded to "Linux" (see [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py#L465)) by PR #26987 and released with azure-cli 2.51.0.

> {
>     "azure-cli": "2.50.0",
>     "azure-cli-core": "2.50.0",
>     "azure-cli-telemetry": "1.0.8",
>     "extensions": {
>         "aks-preview": "0.5.152"
>     }
> }

From the env you provided, I would say you'll need to upgrade your azure-cli to 2.51.0 and take another shot.

**Comment by navba-MSFT** [comment]
@ksingh-scogo The action is pending on you to follow the above suggestion. If you need any further assistance on this issue in future, please feel free to reopen this thread. We would be happy to help.
